src/sentry/integrations/manager.py

```diff
@@ -36,6 +43,9 @@ def exists(self, key: str) -> bool:
 
     def register(self, cls: type[IntegrationProvider]) -> None:
         self.__values[cls.key] = cls
+        if cls.domain:
+            self.__domain_integrations[cls.domain].append(cls.key)
+            self.__integration_domains[cls.key] = cls.domain
 
     def unregister(self, cls: type[IntegrationProvider]) -> None:
         try:
```

Comment:
If you do decide to keep the domain dictionaries around, you'll need to remove integrations from them here.